agent/eni/networkutils/utils_windows.go

```diff
@@ -0,0 +1,117 @@
+// +build windows
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package networkutils
+
+import (
+	"context"
+	"net"
+	"time"
+
+	apierrors "github.com/aws/amazon-ecs-agent/agent/api/errors"
+	"github.com/aws/amazon-ecs-agent/agent/eni/gonetwrapper"
+	"github.com/aws/amazon-ecs-agent/agent/utils/retry"
+	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+// Interface for NetworkUtils
+// The methods declared in this package may or may not add any additional logic over the actual networking api calls.
+// Moreover, we will use a wrapper over Golang's net package. This is done to ensure that any future change which
+// requires a package different from Golang's net, can be easily implemented by changing the underlying wrapper without
+// impacting watcher
+type WindowsNetworkUtils interface {
+	GetInterfaceMACByIndex(int, context.Context, time.Duration) (string, error)
+	GetAllNetworkInterfaces() ([]net.Interface, error)
```

Comment:
Why do you need two similar interfaces? Can't we do with just one interface.

Reply:
gonetwrapper/netutils are wrappers over underlying packages which provide network related functions.
NetworkUtils package adds application specific logic over these functions such as backoff and retry.

Therefore we would need two interfaces